### PR TITLE
feat(open-payments): get outgoing payment

### PR DESCRIPTION
**Pull request by mkurapov**
## Changes proposed in this pull request

- Method for getting outgoing payment 

## Context
 
- Work towards #663 

## Checklist

- [x] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [ ] Documentation added
- [x] Make sure that all checks pass

### CODE REVIEW

File: packages/open-payments/src/client/outgoing-payment.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    throw new Error(
+      'Asset code or asset scale of sending amount does not match up sent amount'
+    )
+  }
+  if (BigInt(sendAmount.value) < BigInt(sentAmount.value)) {
+    throw new Error('Amount sent is larger than maximum amount to send')
+  }
+  if (sendAmount.value === sentAmount.value && payment.failed) {
+    throw new Error('Amount to send matches sent amount but payment failed')
+  }
```
Comment (mkurapov) [inline packages/open-payments/src/client/outgoing-payment.ts:77]: Is this a legitimate concern to check for?

Comment (wilsonianb) [reply packages/open-payments/src/client/outgoing-payment.ts:77]: The check seems valid to me.

File: packages/open-payments/src/client/outgoing-payment.test.ts [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        {
+          url: `${baseUrl}/outgoing-payment`,
+          accessToken: 'accessToken'
+        },
+        openApiValidators.successfulValidator
+      )
+      expect(result).toStrictEqual(outgoingPayment)
+    })
+
+    test('throws if outgoing payment does not pass validation', async (): Promise<void> => {
```
Comment (wilsonianb) [inline packages/open-payments/src/client/outgoing-payment.test.ts:64]: What do you think about adding another test with:
```
      const outgoingPayment = mockOutgoingPayment()
```
and
```
         openApiValidators.failedValidator
```

File: packages/open-payments/src/client/outgoing-payment.test.ts [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        sentAmount: {
+          assetCode: 'USD',
+          assetScale: 2,
+          value: '0'
+        }
+      })
+
+      expect(validateOutgoingPayment(outgoingPayment)).toStrictEqual(
+        outgoingPayment
+      )
```
Comment (wilsonianb) [inline packages/open-payments/src/client/outgoing-payment.test.ts:118]:
```suggestion
      const outgoingPayment = mockOutgoingPayment()

      expect(validateOutgoingPayment(outgoingPayment)).toStrictEqual(
        outgoingPayment
      )
```

File: packages/open-payments/src/client/outgoing-payment.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+export const validateOutgoingPayment = (
+  payment: OutgoingPayment
+): OutgoingPayment => {
+  const { sendAmount, sentAmount } = payment
+  if (
+    sendAmount.assetCode !== sentAmount.assetCode ||
+    sendAmount.assetScale !== sentAmount.assetScale
+  ) {
+    throw new Error(
+      'Asset code or asset scale of sending amount does not match up sent amount'
```
Comment (wilsonianb) [inline packages/open-payments/src/client/outgoing-payment.ts:69]:
```suggestion
      'Asset code or asset scale of sending amount does not match sent amount'
```

**Review by wilsonianb — APPROVED** [review]

**Review by wilsonianb — APPROVED** [review]

**Pull request merged**